### PR TITLE
Add `preloadFunction` option

### LINKED ISSUE #48: Preload script before `goto`
```js
if(options.preload){
  await page.evaluateOnNewDocument(options.preload);
}

await page.goto(...);
```

### PR DESCRIPTION
**Pull request by brandon93s**
Updates from #66;  Fixes #48.

I'll follow with corresponding additions to https://github.com/sindresorhus/capture-website-cli.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -164,6 +164,11 @@ declare namespace captureWebsite {
 		*/
 		readonly isJavaScriptEnabled?: boolean;
 
+		/**
+		Inject a function to be executed prior to navigation; useful for [altering the JavaScript environment](https://github.com/puppeteer/puppeteer/blob/main/docs/api.md#pageevaluateonnewdocumentpagefunction-args).
```
Comment (sindresorhus) [inline index.d.ts:168]: The readme and index.d.ts docs should be in sync.

File: readme.md [hunk 2]
```diff
@@ -474,6 +474,15 @@ Default: `false`
...
+##### preloadFunction
+
+Type: `string | Function`\
+Default: `undefined`
+
+Inject a function to be executed prior to navigation.
+
+This can be useful for [altering the JavaScript environment](https://github.com/puppeteer/puppeteer/blob/main/docs/api.md#pageevaluateonnewdocumentpagefunction-args). For example, you could define a global method on the `window`, overwrite `navigator.languages` to change the language presented by the browser, or mock `Math.random` to return a fixed value.
+
 *This option applies only to `captureWebsite.file()`.*
```
Comment (sindresorhus) [inline readme.md:486]: I think this is misplaced.

Comment (brandon93s) [reply readme.md:486]: 🤦‍♂️

File: index.d.ts [hunk 3]
```diff
@@ -164,6 +164,13 @@ declare namespace captureWebsite {
 		*/
 		readonly isJavaScriptEnabled?: boolean;
 
+		/**
+		Inject a function to be executed prior to navigation.
+
+		This can be useful for [altering the JavaScript environment](https://github.com/puppeteer/puppeteer/blob/main/docs/api.md#pageevaluateonnewdocumentpagefunction-args). For example, you could define a global method on the `window`, overwrite `navigator.languages` to change the language presented by the browser, or mock `Math.random` to return a fixed value.
+		*/
+		readonly preloadFunction?: EvaluateFn;
+
```
Comment (brandon93s) [inline index.d.ts:173]: image

**Comment by brandon93s** [comment]
Documentation has been cleaned up.

**Comment by sindresorhus** [comment]
> I'll follow with corresponding additions to sindresorhus/capture-website-cli.

https://github.com/sindresorhus/capture-website/releases/tag/v1.4.0

**Pull request merged**